### ISSUE TITLE
Questions about the experiments

### ISSUE THREAD

**Issue by lucasdavid** [CVI-SZU/CCAM#8]
Hello! Thank you for sharing this very well organized code.
Additionally, I wonder if you could clarify a few points for me:

- `F.relu` (used in inference_CCAM.py) is commonly used during inference of multi-class CAMs to squash negatively contributing pixels and unrelated pixels together. However, in a binary case, the 0 means uncertain if background or foreground, and it does give you some additional information, no? Why did you choose to go like this, instead of simply checking `ccam > 0.0` or taking the `sigmoid` function and checking for `sigmoid(ccam) > 0.5`?
- a) Which hyper-parameters did you use to train PoolNet? Was all of them set to default?
  b) Did you produce the `.lst` file yourself, pointing out to the locations of the saliency maps produced by inference_CRF.py?
  c) Did you use only images/pseudo saliency maps in `train.txt`, or all files listed in `train_aug.txt`?
- I noticed CCAM has a negative effect on the IoU of classes chair, diningtable and sofa (image). From these, only chair's IoU degeneration is reported in the paper, but not discussed. I guess you ran out of space? Do you have any insights on why that happened?

Thank you very much once again,
cheers.

![image](https://user-images.githubusercontent.com/1595030/185979822-1369cf63-1371-44a6-abc9-2a5e7dce33f7.png)

**Comment by Sierkinhane** [comment]
Hi, I'm participating in a meeting these days! I will reply to your questions later. :-)

**Comment by lucasdavid** [comment]
Of course, at your leisure. Thank you once again.

**Comment by Sierkinhane** [comment]
Hi lucasdavid,
1. Actually the binary case is a special case of multi-class CAMs, so it's reasonable to adopt the commonly used F.relu. I think that the generated activation maps in this way are softer than in other ways like F.sigmoid (based on my observation) and are more suitable to be refined by CRF. I didn't pay much attention to this problem and maybe you can.
2. a) I have uploaded the code for training PoolNet at [here](https://github.com/CVI-SZU/CCAM/tree/master/WSSS/PoolNet) and corresponding configurations are also included.
    b) I didn't generate the .lst file. All you need is to follow this [README](https://github.com/CVI-SZU/CCAM/blob/master/WSSS/README.md) to prepare the background cues in a `.png` format. 
    c) I used `train_aug` set to train PoolNet.
3. There was no enough space to present improvements of 20 classes, so I chose only a half part. Well, I also found that there were no improvements and even a decrease in some of classes like `chair`. I think it's a potential issue in our method C2AM. Because  `chair` and `sofa` are hard classes in PASCAL VOC and the number of images in these classes is smaller than in others, it's hard for C2AM to learn such foregrounds.

**Comment by lucasdavid** [comment]
Thank you very much for all clarifications!
Cheers.
